cdap-api/src/main/java/io/cdap/cdap/api/app/AbstractApplication.java

```diff
@@ -59,6 +59,23 @@ public final void configure(ApplicationConfigurer configurer, ApplicationContext
     configure();
   }
 
+  /*
+   * By default, applications do not support config upgrade. Applications supporting config upgrade should
+   * override this method.
+   */
+  @Override
+  public boolean isUpgradeAllowed() {
```

Comment:
Can we call all "upgrade" as "update" in programmatic API? It is more flexible.

Reply:
Also, this method should be called `isUpdateSupported`

Reply:
Also, move the default implementation to the Application interface to make it backward compatible.

Reply:
Changed it to "update" from upgrade.